homeassistant/components/azure_devops/sensor.py

```diff
@@ -0,0 +1,147 @@
+
+
+class AzureDevOpsSensor(AzureDevOpsDeviceEntity):
+    """Defines a Azure DevOps sensor."""
+
+    def __init__(
+        self,
+        client: DevOpsClient,
+        organization: str,
+        project: str,
+        key: str,
+        name: str,
+        icon: str,
+        measurement: str = "",
+        unit_of_measurement: str = "",
+    ) -> None:
+        """Initialize Azure DevOps sensor."""
+        self._state = None
+        self._attributes = None
+        self._available = False
+        self._unit_of_measurement = unit_of_measurement
+        self.measurement = measurement
+        self.client = client
+        self.organization = organization
+        self.project = project
+        self.key = key
+
+        super().__init__(organization, project, name, icon)
+
+    @property
+    def unique_id(self) -> str:
+        """Return the unique ID for this sensor."""
+        return "_".join([DOMAIN, self.organization, self.key])
+
+    @property
+    def state(self) -> str:
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def device_state_attributes(self) -> object:
+        """Return the attributes of the sensor."""
+        return self._attributes
+
+    @property
+    def unit_of_measurement(self) -> str:
+        """Return the unit this state is expressed in."""
+        return self._unit_of_measurement
+
+
+class AzureDevOpsLatestBuildSensor(AzureDevOpsSensor):
+    """Defines a Azure DevOps card count sensor."""
+
+    def __init__(
+        self, client: DevOpsClient, organization: str, project: str, build: DevOpsBuild
+    ):
+        """Initialize Azure DevOps sensor."""
+        self.build: DevOpsBuild = build
+        super().__init__(
+            client,
+            organization,
+            project,
+            f"{build.project.id}_{build.definition.id}_latest_build",
+            f"{build.project.name} {build.definition.name} Latest Build",
+            "mdi:pipe",
+        )
+
+    async def _azure_devops_update(self) -> bool:
+        """Update Azure DevOps entity."""
+        try:
+            build: DevOpsBuild = await self.client.get_build(
+                self.organization, self.project, self.build.id
+            )
+            self._state = build.build_number
+            self._attributes = {
+                "definition_id": build.definition.id,
+                "definition_name": build.definition.name,
+                "id": build.id,
+                "reason": build.reason,
+                "result": build.result,
+                "source_branch": build.source_branch,
+                "source_version": build.source_version,
+                "status": build.status,
+                "url": build.links.web,
+                "queue_time": build.queue_time,
+                "start_time": build.start_time,
+                "finish_time": build.finish_time,
+            }
+            self._available = True
+        except aiohttp.ClientError as exception:
```

Comment:
Please only wrap the line(s) that can raise with the try... except block. Use `else:` if needed.